src/main/java/io/cdap/plugin/WindowAggregationConfig.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import org.apache.parquet.Strings;
+import org.apache.spark.sql.Column;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Config for Window aggregation plugin.
+ */
+public class WindowAggregationConfig extends PluginConfig {
+
+  @Macro
+  @Description("Specifies a list of fields, comma-separated, to partition the data by. At least 1 field must be " +
```

Comment:
We should add what type of fields are supported as partitionFields

Reply:
All data types are supported. Should we list all data types or only add the phrase `All Data Types are supported`